peachjam/views/legislation.py

```diff
@@ -101,5 +102,103 @@ def get_notices(self):
     def get_repeal_info(self):
         return self.object.metadata_json.get("repeal", None)
 
+    def get_friendly_type(self):
+        return self.object.metadata_json.get("type_name", None)
+
     def get_points_in_time(self):
         return self.object.metadata_json.get("points_in_time", None)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["current_object_date"] = self.object.date.strftime("%Y-%m-%d")
+        context["timeline_events"] = self.get_timeline_events()
+        context["friendly_type"] = self.get_friendly_type()
+        return context
+
+    def get_timeline_events(self):
+        events = []
+
+        work = self.object.metadata_json
+
+        points_in_time = self.get_points_in_time()
+        if points_in_time:
+            expressions = {
+                point_in_time["date"]: point_in_time["expressions"][0]
+                for point_in_time in points_in_time
+            }
+
+        assent_date = self.object.metadata_json.get("assent_date", None)
+        if assent_date:
+            events.append(
+                {
+                    "date": (work["assent_date"]),
+                    "event": "assent",
+                }
+            )
+
+        publication_date = self.object.metadata_json.get("publication_date", None)
+        if publication_date:
+            events.append(
+                {
+                    "date": (publication_date),
```

Comment:
The brackets are unnecessary (on the other events, too)

Reply:
Done